printing/printer.ml

```diff
@@ -344,12 +344,12 @@ let pr_compacted_decl env sigma decl =
        (* Force evaluation *)
        let pb = pr_lconstr_env ~inctx:true env sigma c in
        let pb = if isCast c then surround pb else pb in
-       ids, (str" := " ++ pb ++ cut ()), typ
+       ids, (str" :=" ++ spc () ++ pb ++ cut ()), typ
   in
-  let pids = prlist_with_sep pr_comma (fun id -> pr_id id.binder_name) ids in
+  let pids = hov 0 (prlist_with_sep pr_comma (fun id -> pr_id id.binder_name) ids) in
   let pt = pr_ltype_env env sigma typ in
   let ptyp = (str" : " ++ pt) in
```

Comment:
I guess this should be
```suggestion
  let ptyp = (spc () ++ str": " ++ pt) in
```

Reply:
You prefer for long lines when the `:` is on the next line (as in the new tests)? If ok for you, that's ok to me too.

Reply:
On the testcases, it seems debatable because the identifiers are very short, but would they be longer, we'd end up with
```
long_long_identifier : thing indentend
                                       with crazy margin
                                       in the middle
                                       of your screen
```
so yes, I think it's better overall

Reply:
An alternative is to cut after the `:` (like in `pr_rel_decl`) so that there is a choice between:
```
long_long_identifier
  : thing indented
                   with crazy margin
                   in the middle
                   of your screen
```
and
```
long_long_identifier :
  thing indented
                 with crazy margin
                 in the middle
                 of your screen
```

Reply:
Ok, done that

Reply:
All output looks nice to me.